src/polylineencoder.h

```diff
@@ -100,6 +100,9 @@ class PolylineEncoder
     //! Returns the result of encoding of the given polyline.
     static std::string encode(const Polyline &polyline);
 
+    template <typename T>
+    static std::string encodeList(const T *data, size_t list_length, double T::*latitude, double T::*longitude);
```

Comment:
I would be better to call the function `encode` to overload the existing one.
This function isn't that much generic because you still limit it to a particular member function. It fits to your data structures, but not all. The purpose of this code is being as abstract as possible.

Reply:
I can't see the member function you have mentioned, The user can pass two of its data members that are of type `double`, what did you mean by the **_limiting member function_**? 
also, I think you are right about the function name.

Reply:
> I can't see the member function you have mentioned, The user can pass two of its data members that are of type `double`, what did you mean by the **_limiting member function_**? also, I think you are right about the function name.

What if I have a data structure that doesn't have `latitude` member function? Will your code work?

Reply:
It doesn't work with member functions at all, for that we need another overload that takes pointers to member functions that works with `std::invoke` or the plain way, with the **`data member`** of type `T,` when calling the `encodeList`, we can pass pointer to any public data member that has type `double` , for example, I wrote the test with the struct like this:

```c++
    class DummyPoint {
    public:
        DummyPoint(double t_x, double t_y): x(t_x), y(t_y), speed(0), heading(0){}

        double x;
        double y;
        unsigned short speed;
        float heading;

    };
```

and use member `x` and `y` as `latitude` and `longitude` respectively like below:

```c++ 
auto result = encoder.encodeList(dp,3, &DummyPoint::x, &DummyPoint::y);
```
the names for your data structure that represents the `lat/lon` is not important, you just need to pass a pointer to the wanted data member.